Wrappers/Python/ccpi/processors/RingRemoval.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+from scipy.fftpack import fftshift, ifftshift, fft, ifft
+import numpy as np
+import pywt
+from ccpi.framework import DataProcessor, ImageData, AcquisitionData
+
+class RingRemoval(DataProcessor):
+    
+    '''
+        RingRemoval Processor: Removes vertical stripes from a DataContainer(ImageData/AcquisitionData) 
+        the algorithm in https://doi.org/10.1364/OE.17.008567
+
+    '''
+        
+    def __init__(self, decNum, wname, sigma, info = True):
+        
+        '''
+    
+        Parameters
+        ----------
+        decNum : number of wavelet decompositions
+        
+        wname : (str) name of wavelet filter from pywt 
+            Example: 'db1' -- 'db35', 'haar'
+    
+        sigma : Damping parameter in Fourier space.
+        
+        info : Prints ring removal end message 
+        
+        Returns
+        -------
+        ImageData/AcquisitionData
+            Corrected 2D, 3D, multi-spectral 2D, multi-spectral 3D    
+        '''             
+        
+        kwargs = {'decNum': decNum,
+                  'wname': wname,
+                  'sigma': sigma,
+                  'info': info}
+    
+        super(RingRemoval, self).__init__(**kwargs)            
+                          
+        
+    def check_input(self, dataset):
+        if not ((isinstance(dataset, ImageData)) or 
+                (isinstance(dataset, AcquisitionData))):
+            raise Exception('Processor supports only following data types:\n' +
+                            ' - ImageData\n - AcquisitionData')
+        elif (dataset.geometry == None):
+            raise Exception('Geometry is not defined.')
+        else:
+            return True        
+                
+        
+    def process(self, out = None):
+        
+        data = self.get_input()                
+        decNum = self.decNum
+        wname = self.wname
+        sigma = self.sigma
+        info = self.info
+        
+        # acquisition geometry from sinogram
+        geom = data.geometry
+        
+        # get channels, vertical info
+        channels =  geom.channels    
+        vertical = geom.pixel_num_v
+        
+        # allocate datacontainer space
+        corrected_data = 0.*data 
+        out = corrected_data
+        
+        # allocate numpy space
+        tmp_data_array = corrected_data.as_array()
+        
+        # sinogram numpy array
+        data_array = data.as_array()
+        
+        if channels == 1:        
+            if vertical>0:
+                for i in range(vertical):
+                    J = self.xRemoveStripesVertical(data_array[i], decNum, wname, sigma)
```

Comment:
I think this requires the data to have a specific order. Unless you check it in `check_input`

instead of `data_array[i]`, you should use `data.subset(vertical=i).as_array()`

same for `data_corrected`. You should use `data_corrected.fill(J, vertical=i)`

Reply:
corrected